pkg/oras/discover.go

```diff
@@ -15,9 +16,12 @@ func Discover(ctx context.Context, resolver remotes.Resolver, ref, artifactType
 		return ocispec.Descriptor{}, nil, err
 	}
 
-	discoverer, err := resolver.Discoverer(ctx, ref)
-	if err != nil {
-		return ocispec.Descriptor{}, nil, err
+	discoverer, ok := resolver.(interface {
+		Discover(ctx context.Context, desc ocispec.Descriptor, artifactType string) ([]artifactspec.Descriptor, error)
+	})
```

Comment:
You may check this at the beginning of this function to save remtoe calls.

Reply:
Oh yeah good idea